### PR TITLE
Bundle add files to requested datapack. 

### PR DESCRIPTION
**Pull request by jasebell**
- New view (bundle-add)

- New hander to send the update to datastore.

- Activity finished handler to send info message back to user.

### CODE REVIEW

File: src/cljs/witan/ui/controllers/datastore.cljs [hunk 1]
```diff
@@ -694,6 +694,24 @@
   [event {:keys [idx]}]
   (data/swap-app-state! :app/datastore assoc :ds/data-view-subview-idx idx))
 
+
+(defmethod handle
+  :send-basic-collect-request
+  [event {:keys [groups message]}]
+  #_(activities/start-activity!
+     :send-collect-request
+     (data/new-command!)))
```
Comment (acron0) [inline src/cljs/witan/ui/controllers/datastore.cljs]: I think this is a merge relic.

File: src/cljs/witan/ui/strings.cljs [hunk 2]
```diff
@@ -325,6 +327,8 @@
    :string.activity.add-file-to-datapack/completed      "You successfully added file '%s' to datapack '%s'"
    :string.activity.send-collect-request/failed         "You failed to send a collection request for datapack '%s'"
    :string.activity.send-collect-request/completed      "You successfully sent a collection request to %s recipient(s), for datapack '%s'"
+   :string.activity.add-files-to-datapack/failed "You failed to add files to the datapack."
+   :string.activity.add-files-to-datapack/completed "You successfully added files to the datapack."
```
Comment (acron0) [inline src/cljs/witan/ui/strings.cljs:333]: There are already failed and completed strings that look good (`string.activity.add-file-to-datapack/completed`) - can we use these?

File: src/cljs/witan/ui/controllers/datastore.cljs [hunk 3]
```diff
@@ -951,6 +969,16 @@
...
+(defmethod on-activity-finished
+  [:add-collect-files-to-datapack :completed]
+  [{:keys [args]}]
+  (.info js/toastr (:log args)))
+
+(defmethod on-activity-finished
+  [:add-collect-files-to-datapack :failed]
+  [{:keys [args]}]
+  (.info js/toastr (:log args)))
+
```
Comment (acron0) [inline src/cljs/witan/ui/controllers/datastore.cljs]: Once the activity finishes, what happens? What does the user see or where do they go?

File: src/cljs/witan/ui/components/bundle-add.cljs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          [:div
+           (get-string :string/datapack-collect-intro-text)]]
+         [show-files files]
+         (shared/button {:id :datapack-add
+                         :txt :string/datapack-sharing-bundle-add
+                         :class "btn-success"
+                         :prevent? true
+                         :disabled? (empty? @files)}
+                        #(controller/raise! :data/add-collect-files-to-datapack {:added-files @files
+                                                                                 :datapack-id (:id (:route/params (data/get-app-state :app/route)))}))]]])))
```
Comment (acron0) [inline src/cljs/witan/ui/components/bundle-add.cljs]: The whole form needs disabling whilst the activity is in progress, to stop people hammering the button, changing files etc. Also, a message if/when it fails/succeeds.

**Review by acron0 — CHANGES_REQUESTED** [review]

File: src/cljs/witan/ui/controllers/datastore.cljs (hunk 1 above)
Comment (jasebell) [reply src/cljs/witan/ui/controllers/datastore.cljs]: Removed.

File: src/cljs/witan/ui/strings.cljs (hunk 2 above)
Comment (jasebell) [reply src/cljs/witan/ui/strings.cljs:333]: I don't think we can use those as they are formatted strings, C+S bundle add has no detail on the datapack name at this point as it's not loaded in for the view.

Comment (acron0) [reply src/cljs/witan/ui/strings.cljs:333]: You're correct - my bad

**Review by acron0 — APPROVED** [review]

**Pull request merged**